File: apps/web/src/lib/social/event_category.test.ts
import { test } from 'node:test';
import assert from 'node:assert/strict';
import { EVENT_CATEGORIES, isAthleticCategory } from './event_category';

test('EVENT_CATEGORIES lists every category once in picker order', () => {
	assert.deepEqual([...EVENT_CATEGORIES], ['run', 'cycle', 'class', 'social']);
});

test('run and cycle are athletic', () => {
	assert.equal(isAthleticCategory('run'), true);
	assert.equal(isAthleticCategory('cycle'), true);
});

test('class and social are not athletic', () => {
	assert.equal(isAthleticCategory('class'), false);
	assert.equal(isAthleticCategory('social'), false);
});

File: apps/mobile_android/test/event_category_test.dart
import 'package:flutter_test/flutter_test.dart';

import '../lib/event_category.dart';

void main() {
  test('kEventCategories lists every category once in picker order', () {
    expect(kEventCategories, ['run', 'cycle', 'class', 'social']);
  });

  test('run and cycle are athletic', () {
    expect(isAthleticEventCategory('run'), isTrue);
    expect(isAthleticEventCategory('cycle'), isTrue);
  });

  test('class and social are not athletic', () {
    expect(isAthleticEventCategory('class'), isFalse);
    expect(isAthleticEventCategory('social'), isFalse);
  });
}
